rustls/src/crypto/tls12.rs

```diff
@@ -1,4 +1,63 @@
-pub(crate) fn prf(out: &mut [u8], hmac_key: &dyn super::hmac::Key, label: &[u8], seed: &[u8]) {
+use super::hmac;
+use super::ActiveKeyExchange;
+use crate::error::Error;
+
+use alloc::boxed::Box;
+
+/// An instantiation of the TLS1.2 PRF with a specific, implicit hash function.
+///
+/// See the definition in [RFC5246 section 5](https://www.ietf.org/rfc/rfc5246.txt).
+///
+/// See [`PrfUsingHmac`] as a route to implementing this trait with just
+/// an implementation of [`hmac::Hmac`].
+pub trait Prf: Send + Sync {
```

Comment:
Oh, I thought the ordering was OK here. I think `Prf` is the headline type, with `PrfExpander` in support, and then `PrfUsingHmac` (etc) being relatively uninteresting?

Reply:
So in this case there's tension between "headline type" (relevance/importance) and dependencies (which types/traits depend on which other types/traits).

Quoting the style guide for context:

> Within a module, we prefer to order items top-down. This means that items within a module will depend on items defined below them, but not (usually) above them. The idea here is that the public API, with more internal dependencies, will be read (and changed) more often, and putting it closer to the top of the module makes it more accessible.

I tend to resolve this in favor of dependency order, because when reading a module that better helps me understand how things are put together.